esphome/components/online_image/online_image.cpp

```diff
@@ -120,9 +124,14 @@ void OnlineImage::update() {
 
 #ifdef USE_ONLINE_IMAGE_PNG_SUPPORT
   if (this->format_ == ImageFormat::PNG) {
-    this->decoder_ = esphome::make_unique<PngDecoder>(this);
-  }
+    this->decoder_ = make_unique<PngDecoder>(this);
+  } else
```

Comment:
```suggestion
  }
```
this->format can only have one value, so it should behave exactly the same without an "else" chain.

With the #ifdef combination it might cause issues with the else not having a block, depending on which combination of defines is actually compiled in.

This is also only called once per image download, so the extra few cycles that the additional if's would cost should be negligible.